src/modules/adaptors/Morpho/MorphoAaveV3DebtTokenAdaptor.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import { BaseAdaptor, ERC20, SafeTransferLib, Cellar, Registry, Math } from "src/modules/adaptors/BaseAdaptor.sol";
+import { IMorpho } from "src/interfaces/external/Morpho/IMorpho.sol";
+
+/**
+ * @title Morpho Aave V3 debtToken Adaptor
+ * @notice Allows Cellars to interact with Morpho Aave V3 debtToken positions.
+ * @author crispymangoes
+ */
+contract MorphoAaveV3DebtTokenAdaptor is BaseAdaptor {
+    using SafeTransferLib for ERC20;
+    using Math for uint256;
+
+    //==================== Adaptor Data Specification ====================
+    // adaptorData = abi.encode(address underlying)
+    // Where:
+    // `underlying` is the underlying token address this adaptor is working with
+    //================= Configuration Data Specification =================
+    // NOT USED
+    //====================================================================
+
+    /**
+     @notice Attempted borrow would lower Cellar health factor too low.
+     */
+    error MorphoAaveV3DebtTokenAdaptor__HealthFactorTooLow();
+
+    /**
+     * @notice Strategist attempted to open an untracked Morpho loan.
+     * @param untrackedDebtPosition the address of the untracked loan
+     */
+    error MorphoAaveV3DebtTokenAdaptor__DebtPositionsMustBeTracked(address untrackedDebtPosition);
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure override returns (bytes32) {
+        return keccak256(abi.encode("Morpho Aave V3 debtToken Adaptor V 1.0"));
+    }
+
+    /**
+     * @notice The Morpho Aave V3 contract on Ethereum Mainnet.
+     */
+    function morpho() internal pure returns (IMorpho) {
+        return IMorpho(0x33333aea097c193e66081E930c33020272b33333);
+    }
+
+    /**
+     * @notice Minimum Health Factor enforced after every borrow.
+     * @notice Overwrites strategist set minimums if they are lower.
+     */
+    function HFMIN() internal pure returns (uint256) {
+        return 1.05e18;
```

Comment:
Should this be adjustable with proper permissions per Cellar (and perhaps a timelock)? The reason I ask is because I am wondering about scenarios where the health factor for Morpho or the underlying protocol (AAVE) allows situations with lower HF. That said, 1.05 HF is pretty low, so maybe better to leave it as is.

Reply:
Sounds good, we'll talk about it more but it makes sense.